CoffeeAddicts/Common/ManhattanDistance.cs

```
@@ -14,6 +15,8 @@ public class ManhattanDistance : DistanceFactory
         /// <returns></returns>
         public override double GetDistance(Tuple<CartesianCoordinate, CartesianCoordinate> cartesianCoordinateTuple)
         {
+            Validator.ValidateNotNull(cartesianCoordinateTuple, "cartesianCoordinateTuple");
```

Is this validation really needed? In this case, having null checks seems like you guard against bugs (NullReferenceException), not a treatment for a case. You may let it throw a NullReferenceException right away, so you can know exactly that's a bug